src/dndbeyond/base/character.js

```diff
@@ -343,17 +343,21 @@ class Character extends CharacterBase {
         }
     }
 
-    hasClassFeature(name) {
-        return this._class_features.includes(name);
+    hasClassFeature(name, substring=false) {
+        if (substring) return this._class_features.some(f => f.includes(name));
+        else return this._class_features.includes(name);
     }
-    hasRacialTrait(name) {
-        return this._racial_traits.includes(name);
+    hasRacialTrait(name, substring=false) {
+        if (substring) return this._racial_traits.some(f => f.includes(name));
+        else return this._racial_traits.includes(name);
     }
-    hasFeat(name) {
-        return this._feats.includes(name);
+    hasFeat(name, substring=false) {
+        if (substring) return this._feats.some(f => f.includes(name));
+        else return this._feats.includes(name);
     }
     hasAction(name) {
-        return this._actions.includes(name);
+        if (substring) return this.actions.some(f => f.includes(name));
```

Comment:
really, dude? really?
Can't even use the right variable name.
Much disappointed.

Reply:
Fixed, and squashed and everything.